### PR TITLE
Fix epoch/slot calculations for preview

### PR DESCRIPTION
**Pull request by gostkin**

### CODE REVIEW

File: blockchain-source/src/cardano/time.rs [hunk 1]
```diff
@@ -15,27 +16,31 @@ impl Era {
...
         slot_length: 1,
+        epoch_length_seconds: 432000,
     };
 
     pub const SHELLEY_TESTNET: Self = Self {
         first_slot: 1598400,
         start_epoch: 74,
         known_time: 1595967616,
         slot_length: 1,
+        epoch_length_seconds: 432000,
```
Comment (bartfish) [inline blockchain-source/src/cardano/time.rs:27]: It cannot change, so we can leave this hardcoded, right?

**Review by bartfish — COMMENTED** [review]
LGTM, @gostkin this is fix for preview network, right? Is it already tested or will be deployed on preview?

File: blockchain-source/src/cardano/time.rs (hunk 1 above)
Comment (gostkin) [reply blockchain-source/src/cardano/time.rs:27]: yes

**Comment by gostkin** [comment]
> LGTM, @gostkin this is fix for preview network, right? Is it already tested or will be deployed on preview?

the test I added uses real values from preview network. if the test works the code will work on the real network as well

File: blockchain-source/src/cardano/time.rs (hunk 1 above)
Comment (bartfish) [reply blockchain-source/src/cardano/time.rs:27]: alright, thanks for clarifying

**Review by ecioppettini — APPROVED** [review]

**Pull request merged**